### PR TITLE
[postcss-env-function] Add types for postcss-env-function

### PR DESCRIPTION
**Pull request by MysteryBlokHed**
Adds types for the npm package postcss-env-function (<https://www.npmjs.com/package/postcss-env-function>).

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/postcss-env-function/tslint.json
Comment (codershiba) [inline types/postcss-env-function/tslint.json:1]: This file is no longer needed

File: types/postcss-env-function/tslint.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{ "extends": "@definitelytyped/dtslint/dt.json" }
```
Comment (codershiba) [inline types/postcss-env-function/tslint.json:1]: This file is no longer needed

**Review by ghost — APPROVED** [review]

File: types/postcss-env-function/index.d.mts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+import postcssEnvFunction = require("./index");
```
Comment (jakebailey) [inline types/postcss-env-function/index.d.mts:1]: I am sorta shocked that this works. Isn't this an mts file? Shouldn't this be:

```ts
import postcssEnvFunction from "./index.js";
export default postcssEnvFunction
```

To trigger CJS resolution from MJS?

File: types/postcss-env-function/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
```
Comment (jakebailey) [inline types/postcss-env-function/tsconfig.json:3]: Oh, here's part of the problem; set this to node16.

File: types/postcss-env-function/index.d.mts (hunk 2 above)
Comment (jakebailey) [reply types/postcss-env-function/index.d.mts:1]: Wow, still passes in node16. Confusing.

Comment (jakebailey) [reply types/postcss-env-function/index.d.mts:1]: Still passes, wat

Comment (MysteryBlokHed) [reply types/postcss-env-function/index.d.mts:1]: Not really sure what’s happening either, but I can still replace the contents with the snippet you suggested

**Review by jakebailey — APPROVED** [review]
Well, in the absence of errors, I think this looks fine.

The only additional thing I would suggest is adding a test which `await import`s `post-css-function` to test the ESM API, or maybe an .mts file.

File: types/postcss-env-function/postcss-env-function-tests.ts [hunk 4]
```diff
@@ -1,6 +1,9 @@
 import postcss = require("postcss");
 import postcssEnvFunction = require("postcss-env-function");
 
+// Test ESM import
+(async () => await import("postcss-env-function"))();
```
Comment (jakebailey) [inline types/postcss-env-function/postcss-env-function-tests.ts:5]: I meant more actually testing the types, but, I guess 😄

**Review by jakebailey — APPROVED** [review]

**Comment by MysteryBlokHed** [comment]
Ready to merge

**Pull request merged**